test/extensions/filters/http/wasm/wasm_filter_stress_test.cc

```diff
@@ -167,6 +167,114 @@ TEST_P(GrpcWasmStressTest, CalloutHappyPath) {
   EXPECT_EQ(counters["test_callout_successes"], requests_to_send);
 }
 
+TEST_P(GrpcWasmStressTest, CalloutErrorResponse) {
+  constexpr uint32_t connections_to_initiate = 30;
+  constexpr uint32_t requests_to_send = 30 * connections_to_initiate;
+  const std::string wasm_file = absl::StrCat(
+      TestEnvironment::runfilesDirectory(),
+      "/test/extensions/filters/http/wasm/test_data/grpc_callout_", wasmLang(), ".wasm");
+  // Must match cluster name in the wasm bundle:
+  const std::string callout_cluster_name{"callout_cluster"};
+
+  //
+  // Configure the wasm filter
+  //
+
+  config_helper_.addFilter(fmt::format(R"EOF(
+            name: envoy.wasm
+            config:
+              vm_config:
+                vm: "{}"
+                code:
+                  filename: "{}"
+                allow_precompiled: true
+)EOF",
+                                       wasmVM(), wasm_file));
+
+  //
+  // This test first sends a request from wasm to a callout cluster:
+  //    Client -> Envoy -> Wasm -> Callout Cluster
+  // Envoy then unconditionally forwards the request to the origin cluster
+  //              Envoy -> Origin Cluster
+  //
+  // Both Callout and Origin clusters have a single server that immediately
+  // returns a 200 OK response.
+  //
+
+  addCluster(std::make_unique<ClusterHelper>(StressTest::ORIGIN_CLUSTER_NAME))
+      .addServer(std::make_unique<ServerCallbackHelper>(
+          [](ServerConnection&, ServerStream& stream, Http::HeaderMapPtr&&) {
+            ENVOY_LOG(debug, "Origin server received request");
+            Http::TestHeaderMapImpl response{{":status", "200"}};
+            stream.sendResponseHeaders(response);
+          }));
+  addCluster(std::make_unique<ClusterHelper>(callout_cluster_name))
+      .addServer(std::make_unique<ServerCallbackHelper>(
+          [](ServerConnection&, ServerStream& stream, Http::HeaderMapPtr&&) {
+            ENVOY_LOG(debug, "Callout server received request");
+            ProtobufWkt::Value response;
+            response.set_string_value("response");
+            stream.sendGrpcResponse(Grpc::Status::PermissionDenied, response);
+          }));
+
+  try {
+    bind();
+  } catch (Network::SocketBindException& ex) {
+    if (Network::Address::IpVersion::v6 == ipVersion()) {
+      ENVOY_LOG(info, "Environment does not support IPv6, skipping test");
+      GTEST_SKIP();
+    }
+    throw ex;
+  }
+
+  LoadGeneratorPtr client = start();
+
+  //
+  // Exec test and wait for it to finish
+  //
+
+  Http::HeaderMapPtr request{new Http::TestHeaderMapImpl{
```

Comment:
Good call.  Will fix

Reply:
@silentdai I can't get your suggestion to compile.  I think the issue is that make_unique cannot deduce the type of the initializer list.    Looking at other usages of TestHeaderMapImpl under /test I see tons of usages that look like mine.

So if you come up with a make_unique incantation that actually works I'm happy to adopt it.  Otherwise I think I should do what the rest of the Envoy source code does.